tools/asdf/src/proto.rs

```diff
@@ -0,0 +1,290 @@
+    }
+
+    for line in input.content.lines() {
+        let line = line.trim();
+        // TODO: Handle comments
+        if line.is_empty() {
+            continue;
+        }
+        let (tool, version) = line.split_once(' ').unwrap_or((line, ""));
+        if tool == get_id(None)? {
+            final_version = Some(UnresolvedVersionSpec::parse(version)?);
+            break;
+        }
+    }
+    
+    Ok(Json(ParseVersionFileOutput { version: final_version }))
+}
+
+#[plugin_fn]
+pub fn register_tool(Json(input): Json<RegisterToolInput>) -> FnResult<Json<RegisterToolOutput>> {
+    Ok(Json(RegisterToolOutput {
+        name: input.id,
+        type_of: PluginType::Language,
+        minimum_proto_version: Some(Version::new(0, 46, 0)),
+        default_install_strategy: InstallStrategy::BuildFromSource,
+        plugin_version: Version::parse(env!("CARGO_PKG_VERSION")).ok(),
+		config_schema: Some(schematic::SchemaBuilder::generate::<AsdfPluginConfig>()),
+        ..RegisterToolOutput::default()
+    }))
+}
+
+#[plugin_fn]
+pub fn register_backend(Json(_): Json<RegisterBackendInput>) -> FnResult<Json<RegisterBackendOutput>> {
+    Ok(Json(RegisterBackendOutput {
+        backend_id: get_backend_id()?,
+        source: Some(SourceLocation::Git(GitSource {
+            url: String::from(get_repo()?.url),
+            ..GitSource::default()
+        })),
+        ..RegisterBackendOutput::default()
+    }))
+}
+
+#[plugin_fn]
+pub fn build_instructions(
+    Json(input): Json<BuildInstructionsInput>,
+) -> FnResult<Json<BuildInstructionsOutput>> {
+    let env = get_host_environment()?;    
+
+    if env.os.is_windows() {
+        return Err(PluginError::UnsupportedWindowsBuild.into());
+    }
+    
+    let repo = get_repo()?;
+
+    let git = GitSource {
+        url: repo.url.clone(),
+        // Use default branch
+        reference: None,
+        ..Default::default()
+    };
+
+    let mut instructions = Vec::new();
+
+    let version = input.context.version;
+
+    // Set asdf environment variables
+    let install_download_path = real_path!(buf, input.context.tool_dir).into_os_string().into_string().unwrap();
+    let cores = if env.os.is_mac() {
+        exec_command!("sysctl -n hw.physicalcpu").stdout
+    } else {
+        exec_command!("nproc").stdout
+    };
+    instructions.append(&mut vec![
+        BuildInstruction::SetEnvVar("ASDF_INSTALL_TYPE".into(), "version".into()),
+        BuildInstruction::SetEnvVar("ASDF_INSTALL_VERSION".into(), version.clone().into()),
+        BuildInstruction::SetEnvVar("ASDF_INSTALL_PATH".into(), install_download_path.clone()),
+        BuildInstruction::SetEnvVar("ASDF_DOWNLOAD_PATH".into(), install_download_path.clone()),
+        BuildInstruction::SetEnvVar("ASDF_CONCURRENCY".into(), cores),
+    ]);
+
+    let install_builder_id = get_install_builder_id(version)?;
+    let mut install_instruction = Box::new(BuilderInstruction {
```

Comment:
Since we have the backend now, we can avoid all this builder stuff, just execute the paths like list versions is.